pkgs/tools/networking/cloudflare-warp/default.nix

```
@@ -1,24 +1,30 @@
-{ stdenv, lib, fetchurl, dpkg, autoPatchelfHook, dbus }:
+{ stdenv
+, lib
+, fetchurl
+, dpkg
+, autoPatchelfHook
+, makeWrapper
+, dbus
+, nftables
+}:
 
 stdenv.mkDerivation rec {
   pname = "cloudflare-warp";
-  version = "2022.02.24";
+  version = "2022.7.472";
 
   src = fetchurl {
-    url = "https://pkg.cloudflareclient.com/uploads/cloudflare_warp_2022_2_288_1_amd64_a0be7b47b3.deb";
-    sha256 = "sha256-gBXF0EfFMT6BC6ts/6PQYJH3AAQSDsFoZGK3RZIqmOA=";
+    url = "https://pkg.cloudflareclient.com/uploads/cloudflare_warp_2022_7_472_1_amd64_0e21d8a94d.deb";
+    sha256 = "sha256-AhxmpPpk1twDevppz6ZC3IxE+/Ck8ki47hYa8qAgr8Q=";
   };
 
   nativeBuildInputs = [
     dpkg
     autoPatchelfHook
+    makeWrapper
   ];
 
   buildInputs = [ dbus ];
 
```

Maybe add a desktop file too, since a scheme handler is what it seems to use for team enrollment:

```suggestion
  desktopItems = [
    (makeDesktopItem {
      name = "warp-cli";
      desktopName = "Cloudflare Zero Trust CLI";
      categories = [ "Utility" "Security" "ConsoleOnly" ];
      noDisplay = true;
      mimeTypes = [ "x-scheme-handler/com.cloudflare.warp" ];
      exec = "${pname} teams-enroll-token %u";
      startupNotify = false;
      terminal = true;
    })
  ];
```

Also add `copyDesktopItems` to `nativeBuildInputs`.

---

Why `${pname}`? Wouldn't this call `cloudflare-warp` instead of `warp-cli`. To be honest, I haven't used the teams enrollment feature as shown on the exec.

---

Oops sorry it should be `warp-cli` indeed.

---

Got it. I will work on this right now.

---

@liff so, the package has been updated recently and it didn't include these changes. This PR will only include the module update, so I might do (or if you want, feel free to do it) this change on another PR.

---

#195372